packages/aws-cdk-lib/aws-codepipeline/lib/trigger.ts

```diff
@@ -158,4 +222,28 @@ export class Trigger {
       providerType: this.props.providerType,
     };
   }
+
+  private renderPushFilter(pushFilter?: GitPushFilter[]): CfnPipeline.GitPushFilterProperty[] | undefined {
+    return pushFilter?.map(filter => {
+      const tags: CfnPipeline.GitTagFilterCriteriaProperty | undefined =
+        filter.tagsExcludes?.length || filter.tagsIncludes?.length ? {
+          // set to undefined if empty array because CloudFormation does not accept empty array
+          excludes: filter.tagsExcludes?.length ? filter.tagsExcludes : undefined,
+          includes: filter.tagsIncludes?.length ? filter.tagsIncludes : undefined,
+        } : undefined;
```

Comment:
The original code was as follows. It would be better if `tags` were undefined if there were neither `tagsExcludes` nor `tagsIncludes`, but this is not the case. (The `tags` will be an empty object.)

```ts
        const tags: CfnPipeline.GitTagFilterCriteriaProperty | undefined = {
          // set to undefined if empty array because CloudFormation does not accept empty array
          excludes: filter.tagsExcludes?.length ? filter.tagsExcludes : undefined,
          includes: filter.tagsIncludes?.length ? filter.tagsIncludes : undefined,
        };
```

In other words, this change will result in better code. However, the CloudFormation template will be changed in the following usage cases.

```ts
  triggers: [{
    providerType: codepipeline.ProviderType.CODE_STAR_SOURCE_CONNECTION,
    gitConfiguration: {
      sourceAction: sourceAction,
      // here
      pushFilter: [{
        tagsExcludes: [],
        tagsIncludes: [],
      }],
      // or
      // pushFilter: [{}],
    },
  }],
```

With this CDK code, by the change in this PR, the CFn template would be changed as follows.

```diff
                "Triggers": [
                    {
                        "GitConfiguration": {
                            "Push": [
                                {
-                                   "Tags": {}
                                }
                            ],
                            "SourceActionName": "CodeStarConnectionsSourceAction"
                        },
                        "ProviderType": "CodeStarSourceConnection"
                    }
                ]
```

However, in fact, such a CFn template, i.e., `Tags` is an empty object, will cause the following CFn error and deployments will fail. In other words, this kind of CFn template change cannot happen in a successful stack of existing users. Therefore, there is no need to view this change negatively. (This change case cannot exist in user's environments.)

```
The trigger configuration for the source action is not valid. Make sure to choose one trigger configuration for each source action.
```